### PR TITLE
Add Docker Daemon Socket is Exposed to Containers query for Kubernetes #592

### PR DESCRIPTION
**Pull request by pedro-mimoso**
Sees if Docker Daemon Socket is not exposed to Containers #592

### CODE REVIEW

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package Cx
+
+CxPolicy [ result ] {
+	document := input.document[i]
+    metadata:= document.metadata
```
Comment (joelsou5a) [inline assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:5]: validate if the kind is 'pod'

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	
+
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	   sprintf("metadata.name=%s.spec.volumes.name=%s.hostPath.path", [metadata.name, volumes[c].name]),
+                "issueType":		"IncorrectValue",
+                "keyExpectedValue": sprintf("spec[%s].volumes[%s].hostPath.path is not '/var/run/docker.sock'", [metadata.name, volumes[c].name]),
+                "keyActualValue": 	sprintf("spec[%s].volumes[%s].hostPath.path is '/var/run/docker.sock'", [metadata.name, volumes[c].name])
+              }
+}
```
Comment (joelsou5a) [inline assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:19]: This seems good if it is a pod, but we also should validate the template if it is not a pod and also check the cronjob

**Review by joelsou5a — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego (hunk 2 above)
Comment (joaomartinscx) [reply assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:19]: And to save you some time trying to fix conflicts, I suggest testing these 3 cases separately, and not use a function common to all cases, even though all that changes is the path where the evaluated attribute is.

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	
+
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    document.kind,
+                "issueType":		"IncorrectValue",
+                "keyExpectedValue": "document.kind is not Pod or CronJob",
+                "keyActualValue": 	"document.kind is a Pod or CronJob",
+              }
+}
```
Comment (joaomartinscx) [inline assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:36]: When the resource is neither a Pod nor a CronJob, it's still evaluated, the path where you find the volume hostPath is different, but it's also supposed to check in Resource.Spec.Template.Spec.

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego (hunk 3 above)
Comment (joaomartinscx) [reply assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:36]: @pedro-mimoso Please take a look here, when it's not Pod or CronJob, you're still supposed to check, just in a different path.

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    volumes[c].hostPath.path == "/var/run/docker.sock"
+
+	
+
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	   sprintf("metadata.name=%s.spec.volumes.name=%s.hostPath.path", [metadata.name, volumes[c].name]),
+                "issueType":		"IncorrectValue",
+                "keyExpectedValue": sprintf("spec[%s].volumes[%s].hostPath.path is not '/var/run/docker.sock'", [metadata.name, volumes[c].name]),
+                "keyActualValue": 	sprintf("spec[%s].volumes[%s].hostPath.path is '/var/run/docker.sock'", [metadata.name, volumes[c].name])
```
Comment (joaomartinscx) [inline assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:18]: metadata is at the same level as Spec, not inside

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (joaomartinscx) [inline assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:38]: same thing here

File: assets/queries/k8s/kubernetes_docker_socket_volume/query.rego [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (joaomartinscx) [inline assets/queries/k8s/kubernetes_docker_socket_volume/query.rego:57]: here too

**Review by joaomartinscx — DISMISSED** [review]

**Review by joaomartinscx — COMMENTED** [review]
Everything looks good to me, I recommend approval and merge of this PR.

**Review by joelsou5a — APPROVED** [review]

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**